### PR TITLE
More fixes for broken sibling selector in iOS

### PR DESCRIPTION
**Pull request by sgomes**
This PR adds more workarounds for the general sibling selector (`~`) breakage in iOS, replacing it with the adjacent sibling selector (`+`) or a chain thereof.

@surma @addyosmani PTAL!

### CODE REVIEW

**Comment by addyosmani** [comment]
Initial review LGTM.

**Comment by Garbee** [comment]
I'm slightly worried since this is making the selectors *more strict* it can end up breaking existing implementations. While edge-cases imo, we should keep this in mind in case of any breakage reports.

LGTM otherwise.

**Comment by sgomes** [comment]
@Garbee Yes, some footers that don't fully follow our samples could break, you're right :-/

Sadly, I don't see a workaround for this that doesn't involve stricter rules, and the use case of WebViews in iOS9 is too large for us to ignore (every third party browser or app with web content would have the bug).

File: templates/android-dot-com/styles.css [hunk 1]
```diff
@@ -481,7 +481,8 @@ a img{
     transition: opacity 0.2s cubic-bezier(0.4, 0, 0.2, 1);
   }
 
-  .android-search-box.is-focused ~ .android-mobile-title {
+  .android-search-box.is-focused + .android-navigation-container +
+      .android-mobile-title {
```
Comment (Garbee) [inline templates/android-dot-com/styles.css]: Looks like you are trying to do a double-sibling-selector... [Could be buggy](https://bugs.webkit.org/show_bug.cgi?id=54373). I don't have an iOS device/setup to verify with, but please run [this testcase](http://codepen.io/Garbee/pen/EVvmog) and verify the background of the text is blue and the color is red.

Comment (sgomes) [inline templates/android-dot-com/styles.css]: Thanks for the link, @Garbee! Both the android.com template and your test case look correct. Here's the latter:

image

**Pull request merged**